### DISCUSSION TITLE
No S3 datasource when using S3 File Upload element

### DISCUSSION THREAD

**Discussion by xnohat** [Budibase/budibase#5481] [Help]
Hi team,

I try to using S3 File Upload but there no S3 Data Source to select

![image](https://user-images.githubusercontent.com/4613808/163977818-31621a05-30bc-4a4b-a746-5c0e6366395b.png)

But I already create S3 Data Source

![image](https://user-images.githubusercontent.com/4613808/163977711-ed6676e3-c8a8-42c8-8744-023b9d48fa34.png)

How to using S3 File Upload correctly ?

Budibase version: 1.0.110

**Comment by aptkingston** [comment]
Hey @xnohat. The S3 upload component only works with the real S3 at the moment, not MinIO, which is why it is not listed as an option. Any datasource with a custom endpoint will not be listed in the component.

**Reply by xnohat** [reply]
Thanks for your reply,

I change to official AWS S3 but still cannot get datasource in S3 File Upload

![image](https://user-images.githubusercontent.com/4613808/163987736-1702362b-a495-46f4-bef1-cad75161c873.png)

**Reply by aptkingston** [reply]
LIke I said in my original reply, any datasource with a custom endpoint will not be shown in the component. Please delete the `Endpoint` field and it will appear in the component. We should probably make this more clear in the UI, as I would probably fill it in too.

**Reply by xnohat** [reply]
oh, i got it, thanks @aptkingston

**Comment by Bytelegion** [comment]
@aptkingston in case we are using Oracle S3 Service , is that work with Budibase ?

**Comment by sergey-vin** [comment]
Any luck using this for s3 file uploads? I still see "This Field setting is the wrong data type for this component" even with removed endpoint is s3 storage, and this s3 storage selected in "s3 file upload component".

PS. I use postgresql as a DB and tried text or text[] field / column

**Reply by applicationuser** [reply]
Hello @sergey-vin @xnohat 

I'm having same issue with S3 file upload component , were you able to find a solution ?

S3 data source configured without any endpoint.

**Reply by oscargws** [reply]
> Any luck using this for s3 file uploads? I still see "This Field setting is the wrong data type for this component" even with removed endpoint is s3 storage, and this s3 storage selected in "s3 file upload component".
> 
> PS. I use postgresql as a DB and tried text or text[] field / column

Did either of you figure out how to get this to work?

**Reply by applicationuser** [reply]
@oscargws 
I was able to configure and use s3 file uploads.  We need to create text type column and save the S3 Upload URL  into that text field using save row action.   Please note that 'Field' value in the component would be a unique name not a table field. ( component would not work if you select existing table column)

check this comment.. 
https://github.com/Budibase/budibase/issues/5454#issuecomment-1102481266
